.github/workflows/build-image.yml

```diff
@@ -0,0 +1,48 @@
+name: Build Image
```

Comment:
When does GitHub Action execute?

Reply:
build image and push to Docker Hub when push to main. build image in pull request, 
```yaml
on:
  push:
    branches:
    - main
  pull_request:
    branches:
    - main
```